### ISSUE TITLE
Invariant violation in react-native from `connect`ed component

### ISSUE THREAD

**Issue by AJamesPhillips** [reduxjs/react-redux#1274]
With react 16.7.0, react-redux 7.0.3, redux 4.0.1, and react-native 0.57.0 the following component:

    export default connect(null, null)(() => <View><Text>Test</Text></View>)

Results in an invariant violation similar to this:

"Invariant Violation: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object."

I tried to create a snack here but got a different error: https://snack.expo.io/@ajp/rebellious-salsa

Changing to react-redux 6.0.1 fixes it

**Comment by timdorr** [comment]
You need to be using a newer version of React Native. We return a React.memo() component, which is only valid in newer version of React.

**Comment by AJamesPhillips** [comment]
Thanks for the lightening fast response @timdorr .  I must admit I am pretty confused by react-native versions.  On the Readme I saw it said `As of React Native 0.18, React Redux 5.x should work` and we're on React Native 0.57.0 so that should be fine?  I know the latest version of react-native is 0.59.5.
To further complicate react-native versions, in a different project I had to set the (expo) react-native version to `"https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz"`.  And I'm not clear on what that is yet.
